### PR TITLE
Add Tika extension dev mode test

### LINKED ISSUE #12062: Quarkus Tika's configuration `quarkus.tika.tika-config-path` not work in `quarkus:dev` run
**Describe the bug**
Tika config `quarkus.tika.tika-config-path=tika-config.xml` not work in `quarkus:dev`

**Expected behavior** 
It should work in both `quarkus:dev` and `java -jar` run when set `tika-config-path`

**Actual behavior**
When run with ` ./mvnw quarkus:dev`, it throws below exception:
```java
2020-09-12 17:00:14,107 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:145)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:124)
        at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:346)
        at java.base/java.lang.Class.newInstance(Class.java:604)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:60)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:255)
        ... 17 more
Caused by: io.quarkus.tika.TikaParseException: Invalid tika-config.xml
        at io.quarkus.tika.runtime.TikaRecorder.initializeParser(TikaRecorder.java:33)
        at io.quarkus.tika.runtime.TikaRecorder.initTikaParser(TikaRecorder.java:21)
        at io.quarkus.deployment.steps.TikaProcessor$initializeTikaParser1762155120.deploy_0(TikaProcessor$initializeTikaParser1762155120.zig:88)
        at io.quarkus.deployment.steps.TikaProcessor$initializeTikaParser1762155120.deploy(TikaProcessor$initializeTikaParser1762155120.zig:40)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:235)
        ... 17 more
Caused by: io.quarkus.tika.TikaParseException: tika-config.xml can not be found at tika-config.xml
        at io.quarkus.tika.runtime.TikaRecorder.getTikaConfigStream(TikaRecorder.java:55)
        at io.quarkus.tika.runtime.TikaRecorder.initializeParser(TikaRecorder.java:29)
        ... 21 more
```

**To Reproduce**
1. Go to https://code.quarkus.io/ , add `Apache Tika`, then generate the code and import to IDE
2. Create `src/main/resources/tika-config.xml` file with content `<properties></properties>`
3. Add config `quarkus.tika.tika-config-path=tika-config.xml` in `src/main/resources/application.properties`
3. Run `./mvnw quarkus:dev`
4. [To Compare]. Run `./mvnw quarkus:build` then `java -jar ...`, it will work as expected.

**Configuration**
```properties
quarkus.tika.tika-config-path=tika-config.xml
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 ```
Darwin my-mac 19.6.0 Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64 i386 MacBookPro15,1 Darwin - Output of `java -version`: 
 ```
 - GraalVM version (if different from Java): NA 
 - Quarkus version or git rev: `1.7.3.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
 ```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/me/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 14.0.1, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/openjdk-14.0.1.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.6", arch: "x86_64", family: "mac"
 ```
**Additional context**
It seems `TikaRecorder.class.getResourceAsStream(...)` is using a different classloader which not able to read resource files

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #12062

### CODE REVIEW

File: extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java [hunk 1]
```diff
@@ -47,9 +47,10 @@ private static InputStream getTikaConfigStream(TikaConfiguration config, String
         // Load tika-config.xml resource
         InputStream is;
         if (config.tikaConfigPath.isPresent()) {
-            is = TikaRecorder.class.getResourceAsStream(
-                    config.tikaConfigPath.get().startsWith("/") ? config.tikaConfigPath.get()
-                            : "/" + config.tikaConfigPath.get());
+            is = TikaRecorder.class.getResourceAsStream(config.tikaConfigPath.get());
+            if (is == null) {
+            	is = Thread.currentThread().getContextClassLoader().getResourceAsStream(config.tikaConfigPath.get());
```
Comment (gsmet) [inline extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java:52]: I think you should use the TCCL directly here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added one comment + there is a formatting issue:

```
2020-09-13T19:05:11.5261289Z [ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.12.2:validate (default) on project quarkus-tika: File '/home/runner/work/quarkus/quarkus/extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java' has not been previously formatted.  Please format file and commit before running validation! -> [Help 1]
```

File: extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java (hunk 1 above)
Comment (geoand) [reply extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java:52]: :+1:

**Comment by sberyozkin** [comment]
Sorry, I probably changed something in the source after running the build :-)

File: extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java (hunk 1 above)
Comment (sberyozkin) [reply extensions/tika/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java]: Done

**Comment by sberyozkin** [comment]
Not sure why the JDK14 build has been cancelled

**Comment by geoand** [comment]
It does happen from time to time unfortunately...

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**